### PR TITLE
Upgrade action to Node v16 and actions/core to v1.10

### LINKED ISSUE #61: Upgrade Action to use node 16 instead of node 12
Following the recent blog post ["GitHub Actions: All Actions will begin running on Node16 instead of Node12"](https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/) by GitHub, I noticed this action is still running on `node12`:

https://github.com/JS-DevTools/npm-publish/blob/e42e3720bfe21259120218c19fdbfedcf72692bd/action.yml#L58-L60

It should be fairly easy to change this by replacing `using: node12` with `using: node16` - I wouldn't expect anything in this Action's source code to break as a result of that.

One thing to note, by and large I've seen other Actions use a major version bump when releasing this change. The motivation being that older GitHub Actions runners may not support `node16`. See for example the ["What's new" section of `actions/checkout`](https://github.com/actions/checkout/tree/e6d535c99c374d0c3f6d8cd8086a57b43c6c700a#whats-new).

### LINKED ISSUE #67: Deprecation warning for 'set-output'
When using the action as follows:
```yaml
- uses: JS-DevTools/npm-publish@v1
              with:
                  package: ./extra/apps/app-campaigns/package.json
                  token: ${{ secrets.NPM_TOKEN }}
                  access: public
```

We are getting a warning:
```txt
publish The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environ
```

### PR DESCRIPTION
**Pull request by razor-x**
Passing on my fork: https://github.com/rxfork/npm-publish/actions/runs/3820793379

To use this version while waiting for this PR to merge, you can put

```yaml
uses: rxfork/npm-publish@v1`
```

- Catch ENOENT error
- Fix linter errors
- Update Invalid URL assertions
- Fix assertions on messages containing registry
- Add registry to results
- Drop Node 12 support
- Update dependencies
- Remove Node 10 support comment
- Do not fail fast on CI
- Update TypeError assertion
- Update to @vercel/ncc
- Update github actions core
- Run using node 16
- Add updated test for setOutput
- Rebuild

Closes #61, closes #67

### CODE REVIEW

**Comment by razor-x** [comment]
@JamesMessinger Just wondering if you are still actively maintaining this, it's really nice work and very well tested! If you need some more help let me know.

**Comment by konstantintieber** [comment]
Nice work @razor-x 👍 Hope this will be merged.

**Comment by hardillb** [comment]
@rkrauskopf @JamesMessinger could you please let us know if there will be any further development of this action?

It would be great to get some of the PRs merged to remove the warnings.

I fully understand that commitments and priorities change, if you need help then I'm sure some of us that make use of the action would be willing to help out.

If not then thanks for starting the project, but we need to know so that we should be moving to other options going forward e.g. @razor-x's fork.

**Comment by razor-x** [comment]
@hardillb Actually I had already tagged v1 for this: https://github.com/rxfork/npm-publish/releases/tag/v1. This should be a stable tag to use as I don't have plans to make changes to this fork unless we run into more issues or warnings.

I'd be happy to take over as maintainer here and open things up to more maintainers if @rkrauskopf or @JamesMessinger would take notice of this thread. We can try their email and link back here. I'd much prefer to a keep this repo or transfer it rather than move to a fork.

File: .github/workflows/CI-CD.yaml
Comment (mcous) [inline .github/workflows/CI-CD.yaml:53]: Why is Windows being skipped here?

File: package.json
Comment (mcous) [inline package.json:42]: Why was this script removed? It looks like the `"release"` script below relies on it

File: package.json
Comment (mcous) [inline package.json:48]: 🎉

File: src/action/index.ts
Comment (mcous) [inline src/action/index.ts:74]: I think this is acceptable for now, but moving forward I would like to restructure this to using a mocking library like [testdouble](https://github.com/testdouble/testdouble.js) rather than adding test-specific behavior to the actual source code

File: test/fixtures/mocha-hooks.js
Comment (mcous) [inline test/fixtures/mocha-hooks.js:19]: `fs.rm` has a `force` option to replicate `rm -rf`: https://nodejs.org/docs/latest-v16.x/api/fs.html#fspromisesrmpath-options

```suggestion
  await fs.rm(paths.tmp, { recursive: true, force: true });
```

File: src/npm-publish.ts
Comment (mcous) [inline src/npm-publish.ts:38]: I think this is a good change but can it be split into its own PR so this one can focus on getting the build fixed / deprecations removed?

**Review by mcous — COMMENTED** [review]
Howdy! Thanks for your work getting the various issues with this repo sorted out. I'm a new maintainer here, and I'd love to get this work in and released ASAP. I've got a few minor notes and questions; please let me know if I can be of any help!

File: package.json
Comment (razor-x) [reply package.json:42]: It's been so long since I looked at this I can't remember. Maybe an issue I hit when trying to develop on a fork if I had to guess.

File: src/action/index.ts
Comment (razor-x) [reply src/action/index.ts:74]: Yes, I agree but was just being piratical about getting things working again with the existing patterns.

File: src/npm-publish.ts
Comment (razor-x) [reply src/npm-publish.ts:38]: Been a while so I can't remember why I added it here. It should be exposed though so consumers can get the default / final value that is used.

**Comment by mcous** [comment]
@razor-x I just merged #73 to get CI green so this PR can focus on resolving the deprecation issues. Let me know if you'd like me to take over and resolve the merge conflicts to get this in!

**Comment by razor-x** [comment]
@mcous So glad to see this project has an active maintainer again!

I was about to close this PR now, as I cannot commit the time to resolve the new conflicts resulting from your recent changes. Personally, I would have preferred if you have giving me the opportunity to respond to your comments first to merge this contribution, however if you would like to take over the branch and resolve the conflicts we can see what changes are left that may still be valuable to integrate. 

In the meantime, I will shift focus to the feature in my other PR: https://github.com/JS-DevTools/npm-publish/pull/69

**Comment by mcous** [comment]
> Personally, I would have preferred if you have giving me the opportunity to respond to your comments first to merge this contribution, however if you would like to take over the branch and resolve the conflicts we can see what changes are left that may still be valuable to integrate.

My apologies for steamroller! I'll push the conflict resolution up and get this PR in. My company relies on this action to publish about a half-dozen npm packages, and the deprecations will cause those workflows to fail in less than two months, so I was feeling a little stressed and wanted to move as quickly as possible 😅 

I'll make sure to request your review(s) moving forward

**Comment by mcous** [comment]
@razor-x ok JK turns out I can't just push to your branch. Opened a PR instead so we can get this PR synced up... 💫 https://github.com/rxfork/npm-publish/pull/1

File: src/npm-publish.ts
Comment (razor-x) [reply src/npm-publish.ts:38]: I remember now, it's because I was fixing something related to this output and to keep things consistent it would need to pull from `results` everywhere instead of `options`.

https://github.com/JS-DevTools/npm-publish/blob/71869b5f0195f8c289f623103692895dd046d50d/src/cli/index.ts#L37

File: .github/workflows/ci-cd.yml
Comment (razor-x) [inline .github/workflows/ci-cd.yml:24]: ⭐

File: action.yml
Comment (razor-x) [inline action.yml:64]: ⭐

File: package.json
Comment (razor-x) [inline package.json:50]: ⭐

File: src/npm-publish.ts
Comment (razor-x) [inline src/npm-publish.ts:38]: ⭐ But could be moved to other PR.

File: test/specs/action/failure.spec.js
Comment (razor-x) [inline test/specs/action/failure.spec.js:12]: ⭐ I think this is important, but can't remember why anymore.

File: src/action/index.ts
Comment (razor-x) [inline src/action/index.ts:73]: I think I needed this to make the tests backwards compatible with how the new setOutput works. If things are working without it now maybe we don't need it anymore.

**Review by razor-x — COMMENTED** [review]
@mcous I marked some changes with ⭐ that look valuable. Let me know what you think. For the test changes, up to you if think they are still relevant. Let me know and I can clean up the PR to only include the things we want to keep.

File: package.json
Comment (mcous) [reply package.json:43]: I think these scripts were calling out to https://github.com/JS-DevTools/version-bump-prompt. I think it's fine to remove the scripts (and the dev dep below on `@jsdevtools/version-bump-prompt`) because:

- It's redundant with the vanilla `npm version` command
- CI will run tests before allowing a release to happen
- Any package under @jsdevtools is relatively unlikely to see any maintenance

**Review by mcous — APPROVED** [review]
Looks good to me! @razor-x I just realized I have permission to add maintainers so  I'm going to add you, too, based on your maintenance of the working fork while this repo was unmaintained. Plus that way with two of us we're a little more protected from maintenance gaps

**Pull request merged**
